pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiContributors.scala

```diff
@@ -0,0 +1,276 @@
+    maybeContributor match {
+      case None => scribesMap
+      case Some(contributor) =>
+        val workIds = extractNestedWorkIds(n)
+        workIds match {
+          case Nil =>
+            scribesMap + addIdToMap(workId, scribesMap, contributor)
+          case ids =>
+            scribesMap ++ addIdsToMap(scribesMap, contributor, ids)
+
+        }
+
+    }
+  }
+
+  /**
+    * The scribe name can be within the handNote tag directly or within a
+    * persName node with role="src" inside the handNote tag. If there is a persName tag,
+    * we pick that to construct the scribe name. Otherwise we use the text directly inside the handNote tag
+    */
+  private def parseScribeLabel(n: Node) = {
+    val persNameNodes =
+      (n \ "persName").filter(n => (n \@ "role") == "scr").toList
+    (n.attribute("scribe"), persNameNodes) match {
+      case (Some(_), Nil)      => parseLabelFromHandNode(n)
+      case (_, List(persName)) => Right(Some(persName.text.trim))
+      case (_, list @ _ :: _) =>
+        getFromOriginalPersNode(n, list).map { case (label, _) => Some(label) }
+      case (None, Nil) => Right(None)
+    }
+  }
+
+  /** Very annoingly if the scribe is in this form:
+    *  <handNote scope="minor" scribe="Scribe_A"> <locus target="#Wellcome_Batak_36801_1 #Wellcome_Batak_36801_2" >a 2-62, b 2-7; b 8-22</locus> Gorak-gorahan, etc. Southern form of ta.</handNote>
+    *  we have to get the text of the handNote node without getting the text of the locus node.
+    *  We do this by parsing only the direct children of handNote that are text nodes.
+    * For some reason that I can't understand, this only works if we load the handNode
+    * as a XML
+    */
+  private def parseLabelFromHandNode(n: Node) = {
+    Right(
+      Some(
+        XML
+          .loadString(n.toString())
+          .child
+          .collect {
+            case t: Text => t.text
+          }
+          .mkString
+          .trim
+      ))
+  }
+
+  private def addIdsToMap(scribesMap: Map[String, List[Contributor[Unminted]]],
+                          c: Contributor[Unminted],
+                          nodeIds: List[String]) = {
+    nodeIds
+      .map(
+        id => addIdToMap(id, scribesMap, c)
+      )
+      .toMap
+  }
+
+  private def addIdToMap(workId: String,
+                         scribesMap: Map[String, List[Contributor[Unminted]]],
+                         c: Contributor[Unminted]) = {
+    scribesMap.get(workId) match {
+      case None =>
+        (workId, List(c))
+      case Some(list) => (workId, list :+ c)
+    }
+  }
+
+  private def extractNestedWorkIds(n: Node) = {
+    (n \ "locus")
+      .map { locus =>
+        (locus \@ "target").trim.replaceAll("#", "").split(" ")
+      }
+      .toList
+      .flatten
+  }
+
+  private def createContributor(
+    label: String,
+    role: ContributionRole): Result[Contributor[Unminted]] =
+    createContributor(label = label, id = "", isFihrist = false, role = role)
```

Comment:
Possibly daft question: could we just set default arguments for `id` and `isFihrist` on the other `createContributor` method?

Reply:
Not a daft question. I think I just felt icky about putting an empty string as a default value for an id and instead of using options, which would probably be the best option. So I decided to hide it into an overloaded method which is even uglier